pyaedt/edb_core/primitives.py

```diff
@@ -0,0 +1,92 @@
+"""
+This module .
+"""
+
+from pyaedt import aedt_exception_handler
+from pyaedt.edb_core.layout import EdbLayout
+from pyaedt.edb_core.padstack import EdbPadstacks
+
+class Trace(object):
+    """
+    Manage EDB trace
+    """
+    def __init__(self, width="0.0", point_list=None, layer="", net_name=""):
+        if point_list is None:
+            point_list = [[0.0, 0.0], [1e-3, 0.0]]
+        self.point_list = point_list
+        self.width = width
+        self.layer = layer
+        self.net_name = net_name
+        self.layout_core = EdbLayout(self)
+
+    @aedt_exception_handler
+    def place_line(self):
+
+        path = self.layout_core.Shape("polygon", points=self.point_list)
+        edb_path = self.layout_core.create_path(path, self.layer, self.width, self.net_name, start_cap_style="Flat",
+                                                   end_cap_style="Flat")
+        return edb_path
+
+class ViaDefinition(object):
+    def __init__(self, name="via_def", hole_diam="", pad_diam="", anti_pad_diam="", start_layer="top",
+                 stop_layer="bottom", antipad_shape="Circle", x_size="", y_size="", corner_rad=""):
+        self.name = name
+        self.hole_diam = hole_diam
+        self.pad_diam = pad_diam
+        self.anti_pad_diam = anti_pad_diam
+        self.start_layer = start_layer
+        self.stop_layer = stop_layer
+        self.anti_pad_shape = antipad_shape
+        self.x_size = x_size
+        self.y_size = y_size
+        self.corner_rad = corner_rad
+        self.padstack_core = EdbPadstacks(self)
+
+    def add_via_def_to_edb(self):
```

Comment:
same here, it is a class with single function

Reply:
no I started a draft and committed by mistake